### PR TITLE
[Improve] [Flink-Kubernetes-V2] Add test examples for watching restSvcEndpointSnaps

### PR DESCRIPTION
**Pull request by caicancai**
## Brief change log

There is currently no test example for monitoring restSvcEndpointSnaps in the test, so I added a test example for monitoring restSvcEndpointSnaps

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by caicancai** [comment]
cc @Al-assad  @wolfboys

**Comment by caicancai** [comment]
test success

File: streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/test/scala/org/apache/streampark/flink/kubernetes/v2/example/UsingOperator.scala [hunk 1]
```diff
@@ -112,7 +112,30 @@ class UsingOperator extends AnyWordSpecLike with BeforeAndAfterAll {
...
+      flinkConfiguration = Map("taskmanager.numberOfTaskSlots" -> "2"),
+      jobManager = JobManagerDef(cpu = 1, memory = "1024m", "2G"),
+      taskManager = TaskManagerDef(cpu = 1, memory = "1024m", "2G"),
+      job = JobDef(
+        jarURI = "local:///opt/flink/examples/streaming/StateMachineExample.jar",
+        parallelism = 1
+      )
+    )
+    for {
+      _ <- FlinkK8sOperator.deployApplicationJob(114514, spec)
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/test/scala/org/apache/streampark/flink/kubernetes/v2/example/UsingOperator.scala:134]: Change it to:
 ```scala
for {
       _ <- FlinkK8sOperator.deployApplicationJob(114514, spec)
      // subscribe job status
      _ <- FlinkK8sObserver.evaluatedJobSnaps.flatSubscribeValues().debugPretty.runDrain.fork
      _ <- FlinkK8sObserver.restSvcEndpointSnaps.flatSubscribeValues().debugPretty.runDrain.fork
      _ <- ZIO.never
   }
```

**Review by Al-assad — CHANGES_REQUESTED** [review]

**Comment by caicancai** [comment]
cc @Al-assad @wolfboys

File: streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/test/scala/org/apache/streampark/flink/kubernetes/v2/example/UsingOperator.scala (hunk 1 above)
Comment (caicancai) [reply streampark-flink/streampark-flink-kubernetes-v2/streampark-flink-kubernetes-core/src/test/scala/org/apache/streampark/flink/kubernetes/v2/example/UsingOperator.scala:134]: done

**Review by Al-assad — APPROVED** [review]
LGTM

**Pull request merged**